src/coreclr/jit/targetamd64.h

```diff
@@ -207,32 +207,69 @@
+  // and which may require unique encoding requirements or be generally unavailable
+  // so these end up as the last of their sequences. In particular, esp is used for
+  // stack indexing and is never an option for general LSRA. Both esp and r12 require
+  // an extra byte to encode when used for addressing. ebp is not always available
+  // since its sometimes used for special frame requirements. Both ebp and r13 require
+  // explicit displacement to be encoded when used for certain addressing modes, which
+  // adds 4 bytes to most usages.
+  //
+  // For simd registers, the numerical order is xmm0-xmm7. You then have xmm8-xmm15
+  // which take an additional byte to encode and can also have xmm16-xmm31 for EVEX
+
 #ifdef UNIX_AMD64_ABI
-  #define REG_VAR_ORDER          REG_EAX,REG_EDI,REG_ESI, \
-                                 REG_EDX,REG_ECX,REG_R8,REG_R9, \
-                                 REG_R10,REG_R11,REG_EBX,REG_ETW_FRAMED_EBP_LIST \
-                                 REG_R14,REG_R15,REG_R12,REG_R13
+  #define REG_VAR_ORDER_CALLEE_TRASH    REG_EAX,REG_ECX,REG_EDX,REG_ESI,REG_EDI,REG_R8,REG_R9,REG_R10,REG_R11
+  #define REG_VAR_ORDER_CALLEE_SAVED    REG_EBX,REG_ETW_FRAMED_EBP_LIST REG_R14,REG_R15,REG_R12,REG_R13
+
+  #define REG_VAR_ORDER_FLT_CALLEE_TRASH    REG_XMM0,REG_XMM1,REG_XMM2,REG_XMM3,REG_XMM4,REG_XMM5,REG_XMM6,REG_XMM7,   \
+                                            REG_XMM8,REG_XMM9,REG_XMM10,REG_XMM11,REG_XMM12,REG_XMM13,REG_XMM14,       \
+                                            REG_XMM15
+  #define REG_VAR_ORDER_FLT_CALLEE_SAVED
+
+
+  #define REG_VAR_ORDER_FLT_EVEX_CALLEE_TRASH   REG_VAR_ORDER_FLT_CALLEE_TRASH,REG_XMM16,REG_XMM17,REG_XMM18,REG_XMM19,\
+                                                REG_XMM20,REG_XMM21,REG_XMM22,REG_XMM23,REG_XMM24,REG_XMM25,REG_XMM26, \
+                                                REG_XMM27,REG_XMM28,REG_XMM29,REG_XMM30,REG_XMM31
+  #define REG_VAR_ORDER_FLT_EVEX_CALLEE_SAVED
 #else // !UNIX_AMD64_ABI
-  #define REG_VAR_ORDER          REG_EAX,REG_EDX,REG_ECX, \
-                                 REG_R8,REG_R9,REG_R10,REG_R11, \
-                                 REG_ESI,REG_EDI,REG_EBX,REG_ETW_FRAMED_EBP_LIST \
-                                 REG_R14,REG_R15,REG_R12,REG_R13
+  #define REG_VAR_ORDER_CALLEE_TRASH    REG_EAX,REG_ECX,REG_EDX,REG_R8,REG_R9,REG_R10,REG_R11
+  #define REG_VAR_ORDER_CALLEE_SAVED    REG_EBX,REG_ESI,REG_EDI,REG_ETW_FRAMED_EBP_LIST REG_R14,REG_R15,REG_R12,REG_R13
+
+  #define REG_VAR_ORDER_FLT_CALLEE_TRASH    REG_XMM0,REG_XMM1,REG_XMM2,REG_XMM3,REG_XMM4,REG_XMM5
+  #define REG_VAR_ORDER_FLT_CALLEE_SAVED    REG_XMM6,REG_XMM7,REG_XMM8,REG_XMM9,REG_XMM10,REG_XMM11,REG_XMM12,         \
+                                            REG_XMM13,REG_XMM14,REG_XMM15
+
+  #define REG_VAR_ORDER_FLT_EVEX_CALLEE_TRASH   REG_XMM0,REG_XMM1,REG_XMM2,REG_XMM3,REG_XMM4,REG_XMM5,REG_XMM16,       \
```

Comment:
We want `XMM16-XMM31` to come before the callee saved since the spill and load (2 instructions, 8 bytes minimum) are overall more expensive than the extra byte per instruction